pkg/spire/controller.go

```diff
@@ -297,18 +315,24 @@ func (sc *spireControllerAPIClient) Close() error {
 		if err != nil {
 			return err
 		}
+		sc.serverConn = nil
 	}
 	if sc.workloadAPI != nil {
 		err = sc.workloadAPI.Close()
 		if err != nil {
 			return err
 		}
+		sc.workloadAPI = nil
 	}
 	if sc.workloadConn != nil {
 		err = sc.workloadConn.Close()
 		if err != nil {
 			return err
 		}
+		sc.workloadConn = nil
 	}
+	sc.entryClient = nil
 	return nil
 }
+
+var _ ControllerAPIClient = (*spireControllerAPIClient)(nil)
```

Comment:
I have some questions:
- what is this doing?
- why change from `spireControllerAPIClient` to `ControllerAPIClient`?
- why the blank identifier?

Reply:
This is doing a compile time assert that spireControllerAPIClient struct implements the interface ControllerAPIClient.

This is useful in the case when the interface ControllerAPIClient is changed, the struct spireControllerAPIClient should also be changed and this check will catch this at compile time.

"Interface checks" in https://go.dev/doc/effective_go#interface_conversions